src/app/routes/verifiable-presentation/received-presentations/received-presentations.component.ts

```diff
@@ -1,15 +1,25 @@
-import { Component } from '@angular/core';
-import { MatTableDataSource } from '@angular/material/table';
+import { Component, Input } from '@angular/core';
 
 @Component({
   selector: 'app-received-presentations',
   templateUrl: './received-presentations.component.html',
   styleUrls: ['./received-presentations.component.scss'],
 })
 export class ReceivedPresentationsComponent {
-  displayedColumns: string[] = ['namespace', 'reason'];
-  dataSource = new MatTableDataSource([
-    { namespace: 'installer', reason: 'asdf dsa' },
-    { namespace: 'installer2', reason: 'Some other reason' },
-  ]);
+  @Input() tableData; 
```

Comment:
It's better to have MatTableData object in the same component that displays it. You can pass here regular array and change it to the MatTableData type. There are 2 options:
1. In ngOnInit assign it. -> ngOnInit() { if(tableData) { ... }
2. create a setter -> @Input set tableData(value: SomeInterface) { ... }
This way it will be easier to test it, you do not care about creating MatTableData object in tests, you're just passing array.

Reply:
Thanks for this!